File: tests/samples/lstinputlisting_test/example_code.py
#!/usr/bin/env python3
"""Example Python script for testing lstinputlisting."""

def factorial(n):
    """Calculate factorial recursively."""
    if n <= 1:
        return 1
    return n * factorial(n - 1)

def fibonacci(n):
    """Generate Fibonacci sequence."""
    a, b = 0, 1
    for _ in range(n):
        yield a
        a, b = b, a + b

if __name__ == "__main__":
    print(f"5! = {factorial(5)}")
    print(f"First 10 Fibonacci: {list(fibonacci(10))}")

File: tests/samples/lstinputlisting_test/example_code.js
// Example JavaScript code for testing lstinputlisting
const fibonacci = (n) => {
    if (n <= 1) return n;
    return fibonacci(n - 1) + fibonacci(n - 2);
};

class Calculator {
    constructor() {
        this.result = 0;
    }

    add(x) {
        this.result += x;
        return this;
    }

    multiply(x) {
        this.result *= x;
        return this;
    }
}

// Usage
console.log(`fib(10) = ${fibonacci(10)}`);
const calc = new Calculator();
console.log(calc.add(5).multiply(3).result); // 15
